### PR TITLE
Login unit test #43

### PR DESCRIPTION
**Pull request by grleachman**
@hishamco, @csharpfritz 

Notice I've had to modify the `AuthenticateEventArgs` to accommodate validation in the EventHandler, by adding the `LoginModel`, to facilitate `OnAuthentication` event being handled in the code behind on the test form.

* split out test code from code behind.
* unquoted some params
* added ID param.
* authenticates on submit

Regards
Gareth

### CODE REVIEW

**Comment by hishamco** [comment]
I already started a PR on #102, I though you add commits directly instead of creating a new PR

Let me check this one ..

File: src/BlazorWebFormsComponents/LoginControls/AuthenticateEventArgs.cs [hunk 1]
```diff
@@ -1,11 +1,12 @@
 ﻿using System;
+using static BlazorWebFormsComponents.LoginControls.Login;
 
 namespace BlazorWebFormsComponents.LoginControls
 {
   public class AuthenticateEventArgs : EventArgs
 	{
-
 		public bool Authenticated { get; set; }
+		public LoginModel loginModel { get; set; }
```
Comment (hishamco) [inline src/BlazorWebFormsComponents/LoginControls/AuthenticateEventArgs.cs]: I don't like this change, the intend was to simulate the WebForms controls

**Comment by hishamco** [comment]
FYI I just updated my PR to make everything work as expected without changing the `AuthenticateEventArgs`

File: src/BlazorWebFormsComponents/LoginControls/AuthenticateEventArgs.cs (hunk 1 above)
Comment (grleachman) [reply src/BlazorWebFormsComponents/LoginControls/AuthenticateEventArgs.cs]: My intention was merely to demonstrate how I would test. Glad you were able to leverage that, and implement your ideal solution :)

**Comment by grleachman** [comment]
This PR can be closed, without merging.

**Pull request closed without merging**